### PR TITLE
Add committer info to repo-init-ui commit

### PR DESCRIPTION
**Pull request by smg247**
Without this info the commit fails.

I have also refactored `api.go` based on @droslean's suggestion in https://github.com/openshift/ci-tools/pull/2672.

### CODE REVIEW

File: cmd/repo-init/api.go [hunk 1]
```diff
@@ -228,10 +234,10 @@ func authHandler(log *logrus.Entry) http.HandlerFunc {
 	}
 }
 
-func clusterProfileHandler(log *logrus.Entry) http.HandlerFunc {
+func (s *server) clusterProfileHandler() http.HandlerFunc {
```
Comment (hongkailiu) [inline cmd/repo-init/api.go:237]: I feel handler funcs should not be attached to the server.

File: cmd/repo-init/repo_manager.go [hunk 2]
```diff
@@ -200,10 +199,19 @@ func commitChanges(message, email, name string) error {
 	if err := bumper.Call(os.Stdout, os.Stderr, "git", "add", "-A"); err != nil {
 		return fmt.Errorf("git add: %w", err)
 	}
-	commitArgs := []string{"commit", "-m", message}
-	if name != "" && email != "" {
-		commitArgs = append(commitArgs, "--author", fmt.Sprintf("%s <%s>", name, email))
+
+	if err := bumper.Call(os.Stdout, os.Stderr, "git", "config", "--local", "user.email", email); err != nil {
```
Comment (hongkailiu) [inline cmd/repo-init/repo_manager.go:203]: Is this what this PR is about?
Can we set this up as a git globe setting, more properly in the image that this tool runs?
We probably did this already in our commonly used os image.

We have many tools run the git-push after git-commit.
I do not see anyone need git-config.
Do you notice the failure in production or locally?

Comment (smg247) [reply cmd/repo-init/repo_manager.go:203]: Yes, this is the actual change in this PR. I thought about setting it in the image, but this way allows for the authorized user themselves to appear as the committer. I actually found the pattern in https://github.com/openshift/ci-tools/blob/master/pkg/github/prcreation/prcreation.go#L139-L148

File: cmd/repo-init/api.go (hunk 1 above)
Comment (smg247) [reply cmd/repo-init/api.go:237]: I may be missing something, but I think that is exactly what @droslean was suggesting I do in https://github.com/openshift/ci-tools/pull/2672#pullrequestreview-882975470.

Comment (hongkailiu) [reply cmd/repo-init/api.go:237]: I get the logger part, but not the handler part: the functions do not use anything from the server's objects.
I guess no big deal except personal taste.

**Pull request merged**